### PR TITLE
Reprojection performance test

### PR DESCRIPTION
**Pull request by Lachei**
Adds vsg query pool classes for gpu benchmarking.
Tested the runtime of accumulation in the raygen shader:
- In the raygen shader: 0.8 ms
- In extra compute shader: 0.2 ms

Before final pull request is made, the accumulation code in the raygen shader should be deleted.

Also a few small bugfixes for the raymiss shader are included which caused fireflies.

### CODE REVIEW

**Comment by Lachei** [comment]
@KnightTec I am currently thinking on letting the code for gpu timing in the application. Later i think i might be an idea to add a timing interface to the pipeline abstract class. What do you think about that? Runtime wise this would add nearly nothing (especially compared to the raytracing pipeline) and having timings for each pipeline (including settings for multiple timing steps if a pipeline has multiple stages) should be very useful for precise timings for scientific research.

I will think again about the currently used query pool for pipeline timing and will very likely remove it, as it is a bit unnecessary in the current main function.

Before this will be merged #31 should be merged.

**Comment by KnightTec** [comment]
> 

A generic way to time the render passes would definitely be useful. We could have some kind of `Profiler `object that can be attached to Pipeline/RenderModules, which could be extended beyond timings, e.g. memory consumption.

**Comment by Lachei** [comment]
Very good idea. I think a profiler class then would mainly be a wrapper for a map with strings as keys (the pipeline itself would then fill the map how it sees fit) with a few reserved keywords for timings, memory consumption etc. The values in the map are then maybe tupels with a number and a string, the string being the unit.
This would be especially useful to have a generic way of dumping the whole Profiler to a text.

File: shaders/accumulator.comp [hunk 1]
```diff
@@ -65,7 +78,7 @@ void main(){
 			vec3 prevNor;
 			prevNor.xy = texture(prevNormal, prevPos.xy).xy;
 			prevNor = vec3(sin(prevNor.x) * cos(prevNor.y), sin(prevNor.x) * sin(prevNor.y), cos(prevNor.x));
-			if(dot(normal, prevNor) > .7f){	//we do have a point which can be reprojected
+			if(true || dot(normal, prevNor) > .7f){	//we do have a point which can be reprojected
```
Comment (KnightTec) [inline shaders/accumulator.comp:81]: `true` probably should not be here?

**Review by KnightTec — COMMENTED** [review]
Compiles and runs well on windows. Slowly getting closer to real time performance on my GTX 1070😁.

**Comment by Lachei** [comment]
Renamed 'DoubleMatrix' to 'CameraMatrices'. The 'true ||' is due to an incorrect decompression for certain normals. As we wanted to change the normal compression anyway, the current code uses this workaround...
For future use however the correct code still remains in the codebase.

**Comment by KnightTec** [comment]
>The 'true ||' is due to an incorrect decompression for certain normals. As we wanted to change the normal compression anyway, the current code uses this workaround... For future use however the correct code still remains in the codebase.

I added a short comment about this to the code.

**Review by KnightTec — APPROVED** [review]

**Pull request merged**